modules/git/repo_compare.go

```diff
@@ -259,3 +263,31 @@ func (repo *Repository) GetDiffFromMergeBase(base, head string, w io.Writer) err
 	}
 	return err
 }
+
+// ReadPatch will check if a diff patch exists and return stats
+func (repo *Repository) ReadPatch(prID int64) (commits string, patch []int) {
+	patchFile := fmt.Sprintf("pulls/%d.patch", prID)
+
+	stdout, err := NewCommand("apply", "--stat", patchFile).RunInDir(repo.Path)
+	if err != nil {
+		logger.Warn("No patch, or could not read patch for %v", patchFile)
+		return "", nil
+	}
+	numFile, totalAddition, totalDeletion, err := parseDiffStat(stdout)
+	if err != nil {
+		logger.Error("Could not parse Diff stat: %d, %d, %d, %v", numFile, totalAddition, totalDeletion, err)
+		return "", nil
+	}
+	patch = append(patch, numFile, totalAddition, totalDeletion)
+	loadPatch, err := ioutil.ReadFile(filepath.Join(repo.Path, patchFile))
+	if err != nil {
+		logger.Error("Could not read file %v, %v", loadPatch, err)
+		return "", nil
+	}
+	logger.Warn(string(loadPatch))
```

Comment:
I think the log message should be clear enough and useful. People may panic if they say a WARNING log with a path in it